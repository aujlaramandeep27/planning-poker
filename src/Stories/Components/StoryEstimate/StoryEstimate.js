import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import {PutStoryEstimate, ResetStoryEstimate} from '../../Actions/StoryEstimateActions';
import {viewMeetings, viewStories} from '../../../Navigation/route-actions';
import {Page} from '../../../Common/Page';
import {StartPollingCurrentStory, StopPollingCurrentStory} from "../../../Meetings/Actions/CurrentStoryActions";

export function StoryEstimate({storyEstimate, currentMeeting, user, currentStory, goToMeetings, goToStories, updateStoryEstimate, resetStoryEstimate, startPollingCurrentStory, stopPollingCurrentStory, match}) {
    // TODO: Move estimations to store
    const estimation = [1, 2, 3, 5, 8, 13];

    useEffect(
        () => {
            startPollingCurrentStory(match.params.meetingId);
            return () => {
                stopPollingCurrentStory();
                resetStoryEstimate();
            };
        },
        [startPollingCurrentStory, stopPollingCurrentStory, resetStoryEstimate, match],
    );

    return (
        <Page title={currentMeeting.name} onBack={goToMeetings}>
            <button className="uk-button uk-button-secondary uk-button-small uk-position-small uk-position-top-right"
                    onClick={() => goToStories(currentMeeting._id)}>Host</button>
            <dl className="uk-description-list">
                <dt># {currentStory.name} :</dt>
            </dl>
            <div className="uk-align-center uk-width-1-1@m">
                {estimation.map((estimate, i) =>
                    <button key={i} onClick={() => updateStoryEstimate(user._id, currentStory._id, estimate)}
                            className={`uk-button uk-margin-small-top uk-width-1-1 uk-inline uk-button-primary ${ storyEstimate === estimate ? 'selected' : '' }`} disabled={currentStory._id === null}>
                        {estimate}
                        <span
                            className={'uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right'}
                            hidden={storyEstimate !== estimate}>Selected</span>
                    </button>)
                }
            </div>
        </Page>
    );
}

function mapStateToProps(state) {
    return {
        storyEstimate: state.storyEstimate,
        currentMeeting: state.currentMeeting,
        user: state.user,
        currentStory: state.currentStory
    }
}

function mapDispatchToProps(dispatch) {
    return {
        goToMeetings: () => dispatch(viewMeetings()),
        goToStories: (meetingId) => dispatch(viewStories(meetingId)),
        updateStoryEstimate: (userId, storyId, estimate) => dispatch(PutStoryEstimate(userId, storyId, estimate)),
        resetStoryEstimate: () => dispatch(ResetStoryEstimate()),
        startPollingCurrentStory: (meetingId) => dispatch(StartPollingCurrentStory(meetingId)),
        stopPollingCurrentStory: () => dispatch(StopPollingCurrentStory()),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(StoryEstimate)
