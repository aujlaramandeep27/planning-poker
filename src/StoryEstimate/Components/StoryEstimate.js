import React, {useEffect} from 'react';
import {connect} from 'react-redux';
import {PutStoryEstimate, ResetStoryEstimate} from '../Actions/StoryEstimateActions';
import {viewMeetings, viewStories} from '../../Navigation/route-actions';
import {Page} from '../../Common/Page';
import {GetCurrentStory, StopCurrentStoryPolling} from "../../CurrentStory/Actions/CurrentStoryActions";

export function StoryEstimate({storyEstimate, currentMeeting, user, currentStory, goToMeetings, goToStories, updateStoryEstimate, resetStoryEstimate, getCurrentStory, stopCurrentStoryPolling, match}) {
    // TODO: Move estimations to store
    const estimation = [1, 2, 3, 5, 8, 13];

    useEffect(
        () => {
            getCurrentStory(match.params.meetingId);
            return () => {
                stopCurrentStoryPolling();
                resetStoryEstimate();
            };
        },
        [getCurrentStory, stopCurrentStoryPolling, resetStoryEstimate, match],
    );

    return (
        <Page title={currentMeeting.name} onBack={goToMeetings}>
            <button className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
                    onClick={() => goToStories(currentMeeting._id)}>Host</button>
            <dl className="uk-description-list">
                <dt># {currentStory.name} :</dt>
                <dd>{currentStory.description}</dd>
            </dl>
            <div className="uk-align-center uk-width-1-1@m">
                {estimation.map((estimate, i) =>
                    <button key={i} onClick={() => updateStoryEstimate(user._id, currentStory._id, estimate)}
                            className={`uk-button uk-margin-small-top uk-width-1-1 uk-inline uk-button-primary ${ storyEstimate === estimate ? 'selected' : '' }`} disabled={currentStory._id === null}>
                        {estimate}
                        <span
                            className={'uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right'}
                            hidden={storyEstimate !== estimate}>Selected</span>
                    </button>)
                }
            </div>
        </Page>
    );
}

function mapStateToProps(state) {
    return {
        storyEstimate: state.storyEstimate,
        currentMeeting: state.currentMeeting,
        user: state.user,
        currentStory: state.currentStory
    }
}

function mapDispatchToProps(dispatch) {
    return {
        goToMeetings: () => dispatch(viewMeetings()),
        goToStories: (meetingId) => dispatch(viewStories(meetingId)),
        updateStoryEstimate: (userId, storyId, estimate) => dispatch(PutStoryEstimate(userId, storyId, estimate)),
        resetStoryEstimate: () => dispatch(ResetStoryEstimate()),
        getCurrentStory: (meetingId) => dispatch(GetCurrentStory(meetingId)),
        stopCurrentStoryPolling: () => dispatch(StopCurrentStoryPolling()),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(StoryEstimate)
