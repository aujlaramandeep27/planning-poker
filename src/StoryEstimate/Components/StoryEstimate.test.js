import { StoryEstimate } from './StoryEstimate';

describe('Container: Story Estimate', () => {
  it('should render Estimate correctly', () => {
    const props = {
      storyEstimate: 1,
      goToMeetings: jest.fn(),
      goToStories: jest.fn(),
      updateStoryEstimate: jest.fn(),
      currentMeeting: jest.fn().mockReturnValue({
        name: 'Mock Meeting Name'
      })
    };

    expect(StoryEstimate(props)).toMatchSnapshot();
  })
});
