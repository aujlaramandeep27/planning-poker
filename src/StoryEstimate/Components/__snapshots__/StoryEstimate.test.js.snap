// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: Story Estimate should render Estimate correctly 1`] = `
<Page
  onBack={[MockFunction]}
  title="mockConstructor"
>
  <HostButton
    onHostClick={[MockFunction]}
  />
  <StoryDescription />
  <div
    className="uk-align-center uk-width-1-1@m"
  >
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button selected"
      onClick={[Function]}
    >
      1
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={false}
      >
        Selected
      </span>
    </button>
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button "
      onClick={[Function]}
    >
      2
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={true}
      >
        Selected
      </span>
    </button>
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button "
      onClick={[Function]}
    >
      3
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={true}
      >
        Selected
      </span>
    </button>
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button "
      onClick={[Function]}
    >
      5
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={true}
      >
        Selected
      </span>
    </button>
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button "
      onClick={[Function]}
    >
      8
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={true}
      >
        Selected
      </span>
    </button>
    <button
      className="uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button "
      onClick={[Function]}
    >
      13
      <span
        className="uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"
        hidden={true}
      >
        Selected
      </span>
    </button>
  </div>
</Page>
`;
